typespec-extension/src/code-model-builder.ts

```diff
@@ -213,8 +214,8 @@ export class CodeModelBuilder {
     this.sdkContext = createSdkContext(context, "@azure-tools/typespec-java");
     if (this.supportsAdvancedVersioning()) {
       this.sdkContext.apiVersion = "all";
-    } else {
-      this.sdkContext.apiVersion = "latest";
+    } else if (this.options["api-version"]) {
+      this.sdkContext.apiVersion = this.options["api-version"];
```

Comment:
Maybe we don't set this at all.

Instead, we add another `api-version=all` to the cases we now have `advancedVersioning`.

Reply:
Updated.